app/models/restore_transfer.rb

```diff
@@ -19,9 +19,9 @@ def self.find_fields
     Set.new [:restore_id]
   end
 
-  def cancel!(reason)
+  def cancel!(reason, detail)
     unless cancelled
-      update!(cancelled: true, cancel_reason: reason)
+      update!(cancelled: true, cancel_reason: reason, cancel_reason_detail: detail)
     end
   end
```

Comment:
This looks clean and simple.  Just checking -- are there any associated bag transfer details to synchronize?  If there are, is this done in a controller?